### ISSUE TITLE
How to use load and save for non-pyglove types

### ISSUE THREAD

**Issue by D-X-Y** [google/pyglove#183]
```
import pyglove as pg
from typing import Annotated, Optional, List, Any

class A(pg.Object):

  x: Annotated[Optional[List[Any]], '124'] = None
  y: Annotated[dict, "124"] = dict()

class B:
  pass

res = A(x=[1, 2], y={'b': B()})
print(res)
```

I want to be able to `pg.save(res, 'res.json')` and successfully reload it via `pg.load('res.json')`. Is there a way to implement?

**Comment by D-X-Y** [comment]
Thanks, Daiyi! Both are very helpful! For the first case, do we also need to implement the `from_json_str` function for `class B`?

**Comment by daiyip** [comment]
No. `pg.to_json_str` (or `pg.Symbolic.to_json_str`) knows how to call a symbolic member's `to_json` to formulate JSON string.

**Comment by D-X-Y** [comment]
In this case, how could it accurately load `B` from the json file?

**Comment by daiyip** [comment]
`pg.load(json_file)` shall just work. Didn't it?

Assuming the input file has the following content if `B` is defined in the __main__ module.

```json
{
   "_type": "__main__.B",
}
```
